velox/vector/ComplexVector.cpp

```diff
@@ -615,25 +642,19 @@ int32_t MapVector::compare(
         BaseVector::toString(),
         otherMap->BaseVector::toString());
   }
-  auto result = compareArrays(
-      *keys_,
-      *otherMap->keys_,
-      IndexRange{rawOffsets_[index], rawSizes_[index]},
-      IndexRange{
-          otherMap->rawOffsets_[wrappedOtherIndex],
-          otherMap->rawSizes_[wrappedOtherIndex]},
-      flags);
+  std::vector<vector_size_t> leftData;
+  std::vector<vector_size_t> rightData;
+  auto leftIndices = sortedKeyIndices(index, toRange(sizeAt(index), leftData));
```

Comment:
sortedKeyIndices method feels very inconvenient to use; can it not just take a row number and return a newly created vector of sorted key indices?

Reply:
Looks like some comments got lost. Repeating one of them here:

----
sortedKeyIndices method feels very inconvenient to use; can it not just take a row number and return a newly created vector of sorted key indices?